src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -82,6 +84,53 @@ fn get_simple_intrinsic(cx: &CodegenCx<'ll, '_>, name: &str) -> Option<&'ll Valu
 }
 
 impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
+    fn is_codegen_intrinsic(
```

Comment:
Why not add the code to `codegen_intrinsic_call`?

Reply:
I need to extract the values from the const u32 operands, and for the non-codegened intrinsics, I didn't like the unnecessary overhead of converting them to LLVM values (`OperandRef`s) and back again for no reason. So `is_codegen_intrinsic` takes the `rustc_middle::mir::Operand`s directly, before the args are converted to the backend.

Plus, for the intrinsics that are not passed through to `codegen_intrinsic_call`, this also avoids all of the other steps between the call to `is_codegen_intrinsic()` and `codegen_intrinsic_call()` in `block.rs`, and all of the prepwork in the implementation of `codegen_intrinsic_call()` prior to `match name {`. That's a lot of code that I can avoid calling by implementing it this way.